### ISSUE TITLE
v2.44.0 contains a breaking change to the response data

### ISSUE THREAD

**Issue by GetKraken** [aws/aws-sdk-js#1476]
Hi, 

I noticed this morning a number of my scripts that use the sdk were broken. 

I tracked this down to a breaking change in the SDK at v2.44.0.

At v2.43.0 I could do the following:

``` javascript
var cloudFront = new AWS.CloudFront();
return cloudFront.listCloudFrontOriginAccessIdentities().promise()
    .then(data => {
         console.log(data.Items);
});
```

And data.Items contained the list of items from the response. 

At v2.44.0 Items is undefined which is the breaking API change.

I tracked this down to the following change in lib/request.js:

``` javascript
var data = AWS.util.copy(resp.data);
data.$response = resp;
resolve(data);
```

The copy only copies the properties of the object which means Items which is unpacked from the CloudFrontOriginAccessIdentityList is undefined. 

I have used listCloudFrontOriginAccessIdentities in my example above but this applies to pretty much all methods due to the result unpacking performed in the SDK. 

I know the documentation does state that the response is CloudFrontOriginAccessIdentityList (map) containing Items for the example above but because these properties were exposed at the root of the object we used those instead of going via the CloudFrontOriginAccessIdentityList property. 

Hence, some of our use of the SDK has been broken by this change. 

Also, in addition the response object seems to have circular dependencies in it which means all the JSON.stringify of the response from the SDK throws errors. As I log some of the responses from the SDK as part of diagnostic logging this means the stringify failed. I think this is also a breaking change in v2.44.0.

For us this will force us to go through all our services and hook the version to v2.43.0 to ensure this doesnt break in an uncontrolled way.

**Comment by northerncodemky** [comment]
Related to #1474  (circular dependencies)

**Comment by chrisradek** [comment]
@GetKraken 
We have a fix in master for the circular references issue that we planned to release today.

I have confirmed the other issue you're seeing is also a bug. Operations whose response has a [`payload` member](https://github.com/aws/aws-sdk-js/blob/master/apis/cloudfront-2017-03-25.normal.json#L2371) are affected. Accessing fields of the response works if following what's in our documentation, but we also 'unpack' the results for backwards-compatibility reasons. We will have to address this as well.

**Comment by chrisradek** [comment]
I've put up a PR for a fix, #1477. Apologies for the trouble caused by this bug.

**Comment by northerncodemky** [comment]
@chrisradek Thanks for fixing that one so quickly. Verified that resolves the issue we were facing and can now use 2.45.0.
